gateway/src/main/java/com/mx/path/gateway/configuration/Configurator.java

```diff
@@ -121,6 +121,8 @@ public final T buildGateway(ObjectMap map, String clientId) {
 
     getObserver().notifyClientGatewayInitialized(clientId, gateway);
 
+    gateway.getBaseAccessor().afterInitialize(this, observer);
```

Comment:
It's been a minute since I've looked at this code... I think this code should notify the observer and not pass that responsibility to the callback.

Reply:
Also, this is not the place to do an accessor afterInitialize notification. That should be done once for every accessor. This will only do it once after all have been initialized. This is basically duplicating the `notifyClientGatewayInitialized`

Reply:
The observer also has an `notifyGatewayInitialized` method. Would you not say that we already have a function that is doing what is being request on https://mxcom.atlassian.net/browse/MC-40?

Reply:
I think the intention here is different. We want to be able to implement a callback in any accessor that will be called once after it is configured and initialized. That means that when we configure a "gateway" many accessorInitialized methods could be invoked. This is definitely more tricky that afterGatewayInitialized because of how we stand accessors up (they aren't actually instantiated during system configuration, only their scope proxy is)

An accessor initialized method would be able to do some one-time things like setup maskers, initialize listeners, anything that needs to be done once during system initialization.